hotsos/core/ycheck/engine/properties/requires/types/apt.py

```diff
@@ -28,32 +28,147 @@ def installed_versions(self):
 
         return _versions
 
-    def package_version_within_ranges(self, pkg, versions):
-        result = False
-        versions = sorted(versions, key=lambda i: str(i['min']), reverse=True)
+    def normalize_version_criteria(self, version_criteria):
+        """Normalize all the criterions in a criteria.
+
+        Normalization does the following:
+        - removes empty criteria
+        - replaces old ops with the new ones
+        - sorts each criterion(ascending) and criteria(descending)
+        - adds upper/lower bounds to criteria, where needed
+
+        Args:
+            version_criteria (list): List of version ranges to normalize
+
+        Returns:
+            list: Normalized list of version ranges
+        """
+
+        # Step 0: Remove empty criteria
+        version_criteria = [x for x in version_criteria if len(x) > 0]
+
+        # Step 1: Replace legacy ops with the new ones
+        legacy_ops = {"min": "ge", "max": "le"}
+        for idx, version_criterion in enumerate(version_criteria):
+            for lop, nop in legacy_ops.items():
+                if lop in version_criterion:
+                    version_criterion[nop] = version_criterion[lop]
+                    del version_criterion[lop]
+                    version_criteria[idx] = version_criterion
```

Comment:
i don think this line (57) is needed? since version_criteria is already in version_criterion at position idx?